src/Bicep.Core/Parser/Parser.cs

```diff
@@ -324,14 +365,12 @@ private SyntaxBase FunctionCallOrVariableAccess()
                 if (this.Check(TokenType.RightParen))
                 {
                     // end of function call
-                    // return the accumulated arguments
-                    var closeParen = this.reader.Read();
-                    var argumentNodes = arguments.Select(a => new FunctionArgumentSyntax(a.expression, a.comma));
-                    return new FunctionCallSyntax(identifier, openParen, argumentNodes, closeParen);
+                    // return the accumulated arguments without consuming right paren a caller must consume it
+                    return arguments.Select(a => new FunctionArgumentSyntax(a.expression, a.comma));
```

Comment:
This is actually one place where I think we very explicitly want to force enumeration (don't return a `Select`) directly, use a `.ToList()` or build a list using a conventional loop. You don't want it to be evaluated multiple times, or lazy-evaluated.

I'd actually personally prefer using a regular loop even if it's a little more verbose as it makes the parsing logic easier to follow.

Reply:
Good point, I updated the code to use a loop and returns an immutable array